apps/srt-tunnel.cpp

```diff
@@ -478,6 +504,10 @@ class TcpMedium: public Medium
     // Just models. No options are predicted for now.
     void ConfigurePre(int )
     {
+#if defined(__APPLE__)
+        int optval = 1;
+        setsockopt(m_socket, SOL_SOCKET, SO_NOSIGPIPE, &optval, sizeof(optval));
```

Comment:
Socket to be configured is here passed as argument in this method. I know this doesn't make much sense in this design (it was predicted as a possibility that both listener and accepted socket could be passed here potentially, which eventually didn't happen), but the code should be kept consistent. Actually it doesn't make any sense to pass the socket in `ConfigurePre()` anywhere in this code, and it should work directly on `m_socket` field, but this better be changed everywhere in this file as a general refactoring, or not done anywhere.

This differs to `ConfigurePost`, as this is for options that are not to be derived by the accepted socket from the listener socket - and therefore it is called just after "accept", as well as just after "connect", therefore it needs that the socket ID be passed to it explicitly as an argument. For `ConfigurePre()` indeed no parameter is necessary.

It's up to you if you want to do this general refactoring or not, but if you don't want to do this, then please change in this function the socket passed to `setsockopt` be taken from the `ConfigurePre` argument, not from `m_socket`.

Reply:
It's my failure that had not referred to SRTMedium implementation. Now it has been edited.